File: huffman_coding_scala/src/EqualityCheck.scala
/**
  * Created by zacharymikel on 4/17/17.
  */
class EqualityCheck {

  var statusMessage: String = ""

  def check(s: String, x: String): Boolean = {
    val sArray = s.toCharArray
    val xArray = x.toCharArray
    var i = 0
    while (i < s.length) {
      {
        if (sArray(i) != xArray(i)) {
          statusMessage = "Fail! Doesn't match at index " + i
          return false
        }
      }
      {
        i += 1; i - 1
      }
    }
    statusMessage = "Success! Files match."
    true
  }

  def getStatusMessage: String = { statusMessage }

}

File: huffman_coding_java/src/EqualityCheck.java
/**
 * Created by zacharymikel on 4/12/17.
 */
public class EqualityCheck {

    public static String statusMessage;

    public static boolean check(String s, String x) {

        char[] sArray = s.toCharArray();
        char[] xArray = x.toCharArray();
        for(int i = 0; i < s.length(); i++) {
            if(sArray[i] != xArray[i]) {
                statusMessage = "Fail! Doesn't match at index " + i;
                return false;
            }
        }

        statusMessage = "Success! Files match.";
        return true;
    }

    public static String getStatusMessage() { return statusMessage; }

}
